pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1224,6 +1240,27 @@ spec:
 			"Normal Started",
 			"Warning Failed PipelineRun foo/pipeline-missing-object-param-keys parameters is missing object keys required by Pipeline foo/a-pipeline-with-object-params's parameters: PipelineRun missing object keys for parameters",
 		},
+	}, {
+		name: "invalid-pipeline-array-index-out-of-bound",
+		pipelineRun: parse.MustParseV1beta1PipelineRun(t, `
+metadata:
+  name: pipeline-param-array-out-of-bound
+  namespace: foo
+spec:
+  pipelineRef:
+    name: a-pipeline-with-array-indexing-params
+  params:
+    - name: some-param
+      value:
+        - "a"
+        - "b"
+`),
+		reason:         ReasonParamArrayIndexingInvalid,
+		permanentError: true,
+		wantEvents: []string{
+			"Normal Started",
+			"Warning Failed PipelineRun foo/pipeline-param-array-out-of-bound array indexing params fail validation: failed to validate Pipeline foo/a-pipeline-with-array-indexing-params's parameters: non-existent param references:[$(params.some-param[2]",
```

Comment:
missing closing brackets `)]` in `[$(params.some-param[2]`

Reply:
is it possible to include the length of an array and the index in the error message `non-existent param references:[$(params.some-param[2]`? It will help the users to learn exactly what is the issue.

```
An array index \"2\" is out of bound, parameter \"some-param\" has two elements in \"$(params.some-param[2])\".
```

Reply:
the bracket is missed intentionally otherwise it will fail the test. Since we will read the string as regex pattern, and if I adds bracket it will lead to the error. I don't know if there's a better way

Reply:
> is it possible to include the length of an array and the index in the error message `non-existent param references:[$(params.some-param[2]`? It will help the users to learn exactly what is the issue.
> 
> ```
> An array index \"2\" is out of bound, parameter \"some-param\" has two elements in \"$(params.some-param[2])\".
> ```

That's possible and makes sense. Maybe we could update it after https://github.com/tektoncd/pipeline/pull/6180 is merged?

Reply:
Can you please open an issue to track it?

Reply:
> the bracket is missed intentionally otherwise it will fail the test. Since we will read the string as regex pattern, and if I adds bracket it will lead to the error. I don't know if there's a better way

It looks half baked, how is the error returned to the user when they try to create a pipelineRun?

Reply:
> > the bracket is missed intentionally otherwise it will fail the test. Since we will read the string as regex pattern, and if I adds bracket it will lead to the error. I don't know if there's a better way
> 
> It looks half baked, how is the error returned to the user when they try to create a pipelineRun?

I think this is just for test purpose, the regex function is only used in tests to check the events. Since event is complicated so it uses regex to match the content of the event. The users should be able to see the full events

Reply:
> Can you please open an issue to track it?

Sure!